pkg/main.go

```diff
@@ -66,7 +66,8 @@ func runServer() error {
 	studentStore := postgres.StudentStore{db}
 	observationStore := postgres.ObservationStore{db}
 	schoolStore := postgres.SchoolStore{db}
-	userStore := postgres.UserStore{db}
+	// userStore := postgres.UserStore{db}
+	userHandler := user.NewUserHandler(server, db)
```

Comment:
isn't store also is an implementation detail of the user package? storage detail implementation to be precise, and shouldn't storage detail implementation need no connection whatsoever in this tier? ato gw sotoy

this change is actually to prevent core being passed from outside layer to inner layer. i haven't do the performance test so i don't know the advantages and disadvantages between these 2

Reply:
I don't think the store is an implementation detail of the user package. Currently, data persistent is the responsibility of `postgres` package. `user` package responsibility is to handle the incoming request. 

If the `user` package care about how the data is being persisted, then it will have 2 responsibility right? That's why the store is being created outside of the `user` package.

I think its fine though for main to have to handle store and pass it in to the `user` package. Since It's job is to tie all the pieces of our codebase together.

Reply:
hmm i see, so you suggest di `user` package to be refactor like `store` package?